protected/humhub/modules/comment/models/forms/CommentForm.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace humhub\modules\comment\models\forms;
+
+use humhub\modules\comment\models\Comment;
+use humhub\modules\content\components\ContentActiveRecord;
+use Yii;
+use yii\web\ServerErrorHttpException;
+
+/**
+ * CommentForm
+ * @package humhub\modules\comment\models\forms
+ *
+ * @since 0.5
+ */
+class CommentForm extends yii\base\Model
+{
+    public $message;
+    public $fileList;
+    public $comment;
+
+    /**
+     * @var Comment|ContentActiveRecord The model to comment
+     */
+    public $target;
+
+    public function __construct($target, $fileList = null)
+    {
+        $this->target = $target;
+        $this->fileList = $fileList;
+        parent::__construct();
+    }
+
+    /**
+     * @return array validation rules for model attributes.
+     */
+    public function rules()
+    {
+        return [
+            [['message'], 'required', 'isEmpty' => function ($message) {
+                $hasFile = !is_null($this->fileList) && !empty($this->fileList) ? true : false;
```

Comment:
`is_null` is redundant since `empty` also checks for null values. Also the expression result is a boolean so you do not have to use `?:` here.

Don't want to be too picky but I would do the following:

- We should always have a comment model from the beginning, either a new one or an existing one in case of edit, currently the comment model is set after the save process.
- Within the hasFile validation do a check like the following:

```php
$hasFiles = !empty($this->fileList) || !empty($this->comment->fileManager->find()->count());
```

- Remove the file query from the controller action



$hasFiles = !empty($this->fileList) || $this->target

Reply:
@buddh4  Done! Thanks for the comment! Very useful!